From 6e3db4846a393558324733a1d18fdf9858d543fe Mon Sep 17 00:00:00 2001
From: Adrian Alonso <aalonso00@gmail.com>
Date: Fri, 17 Dec 2010 15:56:45 -0600
Subject: [PATCH 1/2] ml507: add uartlite config options

* Add CONFIG_XILINX_UARTLITE options
* From xparameters find out which serial module
  is included in the hardware project and configure
  accord the available device

Signed-off-by: Adrian Alonso <aalonso@secretlab.ca>
---
 include/configs/ml507.h |   16 +++++++++++++++-
 1 files changed, 15 insertions(+), 1 deletions(-)

diff --git a/include/configs/ml507.h b/include/configs/ml507.h
index 6454ef5..be429b6 100644
--- a/include/configs/ml507.h
+++ b/include/configs/ml507.h
@@ -172,7 +172,18 @@

 #define CFG_MEMTEST_START	0x0400000	/* memtest works on	*/
 #define CFG_MEMTEST_END		0x0C00000	/* 4 ... 12 MB in DRAM	*/
-
+/*
+ *  * Serial port
+ *   * uartLite
+ *    */
+#ifdef XPAR_UARTLITE_0_BASEADDR
+#define CONFIG_XILINX_UARTLITE
+#define CONFIG_SERIAL_BASE      XPAR_UARTLITE_0_BASEADDR
+#define CONFIG_BAUDRATE         XPAR_UARTLITE_0_BAUDRATE
+#define CONFIG_SYS_BAUDRATE_TABLE   { CONFIG_BAUDRATE }
+#define XPAR_UART_0_CLOCK_FREQ_HZ XPAR_PLB_CLOCK_FREQ_HZ
+#else
+#ifdef XPAR_UARTNS550_0_BASEADDR
 /*
  * 16550 UART and console
  */
@@ -183,6 +194,9 @@
 #define CFG_NS16550_REG_SIZE	-4
 #define CFG_NS16550_CLK		XPAR_XUARTNS550_CLOCK_HZ
 #define CFG_NS16550_COM1	(XPAR_UARTNS550_0_BASEADDR + 0x1003)
+#define XPAR_UART_0_CLOCK_FREQ_HZ   XPAR_XUARTNS550_CLOCK_HZ
+#endif
+#endif

 #define CFG_BAUDRATE_TABLE	\
 	{300, 600, 1200, 2400, 4800, 9600, 19200, 38400,115200}
--
1.7.3.3
